aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -272,61 +268,50 @@ mod loader {
 
         /// Override the sleep implementation for this [`ConfigLoader`]. The sleep implementation
         /// is used to create timeout futures.
-        pub fn sleep_impl(mut self, sleep: impl AsyncSleep + 'static) -> Self {
+        pub fn sleep_impl(mut self, sleep: impl IntoShared<SharedAsyncSleep>) -> Self {
```

Comment:
for places where we add `IntoShared` we should probably add examples to the docs because it's not necessarily obvious what will satisfy this trait

Reply:
Good call.